homeassistant/components/tedee/lock.py

```diff
@@ -55,8 +55,13 @@ def __init__(
         super().__init__(lock, coordinator, "lock")
 
     @property
-    def is_locked(self) -> bool:
+    def is_locked(self) -> bool | None:
         """Return true if lock is locked."""
+        if self._lock.state in (
+            TedeeLockState.HALF_OPEN,
```

Comment:
Problem here is, that lock could be in state `half_open` but still be locked, e.g. if you did a half rotation so that'd be confusing

Reply:
But can we tell confidently that it's locked?

Reply:
Depends on the state it was in before. If it was locked before I think it should still be locked, if it was unlocked it is probably still unlocked

Reply:
For my lock it would always be locked while half open, as I calibrated the movement to stop just after opening

Reply:
These are all assumptions. We don't know how far someone rotated the handle.
Example: The lock is locked, and I rotate the handle until it's nearly open. The lock itself is still in the half-open state, but we should not show it locked in HA.
In my opinion, we should not make any assumptions. By marking it as unknown a user can create a automation for it, for example call the lock service if it's unknown

Reply:
fair enough.